### DISCUSSION TITLE
2022-02-21 weekly planning

### DISCUSSION THREAD

**Discussion by kgodey** [mathesar-foundation/mathesar#1084] [Maintainer discussions]
## About
At the start of each week, we start a discussion covering Mathesar's progress, team member check ins, and general updates. Everyone is welcome to add a new topic to discuss or to comment on existing topics.

## Discussion topics
- [Last Week in Mathesar](https://github.com/centerofci/mathesar/discussions/1084#discussioncomment-2210674)
- [Frontend priorities](https://github.com/centerofci/mathesar/discussions/1084#discussioncomment-2210676)
- [Backend priorities](https://github.com/centerofci/mathesar/discussions/1084#discussioncomment-2210678)
- [Design priorities](https://github.com/centerofci/mathesar/discussions/1084#discussioncomment-2210679)
- [Product priorities](https://github.com/centerofci/mathesar/discussions/1084#discussioncomment-2210683)
- [Comms Assignee schedule](https://github.com/centerofci/mathesar/discussions/1084#discussioncomment-2210685)
- [Upcoming Core Team Events](https://github.com/centerofci/mathesar/discussions/1084#discussioncomment-2210686)
- [Dominykas check in](https://github.com/centerofci/mathesar/discussions/1084#discussioncomment-2220007)
- [Brent's check in](https://github.com/centerofci/mathesar/discussions/1084#discussioncomment-2222232)
- [Pavish's check in](https://github.com/centerofci/mathesar/discussions/1084#discussioncomment-2222343)
- [Ghislaine's check in](https://github.com/centerofci/mathesar/discussions/1084#discussioncomment-2223825)
- [Mukesh's check in](https://github.com/centerofci/mathesar/discussions/1084#discussioncomment-22259780)
- [Sean's check in](https://github.com/centerofci/mathesar/discussions/1084#discussioncomment-2228573)
- [Kriti's check in](https://github.com/centerofci/mathesar/discussions/1084#discussioncomment-2232145)

## Weekly check-in
@centerofci/mathesar-core please post your weekly check-in as a top-level response and edit the discussion topics above to add your check in as a topic. Others are also welcome to respond.

Prompts:
- How did your most recent work go?
- What are you planning to work on soon?
- Do you anticipate any dependencies or blockers that someone else on the team can help you with?

If you'd like to add anything else to your check-in that's not related to the prompts, feel free to do so.

**Comment by kgodey** [comment]
## Last Week in Mathesar

### Feedback Requested
- **Discussions**:
  - https://github.com/centerofci/mathesar/discussions/1081
- **Specs**:
  - https://github.com/centerofci/mathesar-wiki/pull/35
  - [DRAFT] https://github.com/centerofci/mathesar-wiki/pull/36

### Complete
- We submitted our Google Summer of Code 2022 organization application!
- **Major wiki updates**:
  - The [Product Spec for Views](https://wiki.mathesar.org/en/product/specs/2022-01-views) is getting a rewrite, but it's not complete yet.
- **Full stack and general work**:
  - https://github.com/centerofci/mathesar/pull/1032
- **Backend work**:
  - https://github.com/centerofci/mathesar/pull/1053
  - https://github.com/centerofci/mathesar/pull/1067
  - https://github.com/centerofci/mathesar/pull/1061
  - https://github.com/centerofci/mathesar/pull/1069
- **Frontend work**:
  - https://github.com/centerofci/mathesar/pull/1075

**Comment by kgodey** [comment]
## Frontend priorities

1. Complete front end changes to https://github.com/centerofci/mathesar/pull/1047
2. Complete #775
3. Complete and merge basic E2E testing: https://github.com/centerofci/mathesar/issues/640. PRs:
	1. https://github.com/centerofci/mathesar/pull/1062
4. Make progress on issues from the [Initial Data Types](https://github.com/centerofci/mathesar/milestone/11) and [2022-02 improvements](https://github.com/centerofci/mathesar/milestone/43) milestones. Some ideas: 
	1. #500
	2. #742
	3. #970
	4. #928
	5. #929
	6. #936
	7. #475

@pavish @seancolsen Please comment below with thoughts/concerns.

**Reply by pavish** [reply]
Looks good to me.

There are currently two open frontend draft PRs, which isn't mentioned above. The goal for this week also includes getting them ready and possibly merged.

* https://github.com/centerofci/mathesar/pull/1076
* https://github.com/centerofci/mathesar/pull/1080

**Reply by seancolsen** [reply]
Looks good.

As noted also in my [check in](https://github.com/centerofci/mathesar/discussions/1084#discussioncomment-2228573), #1076 is on hold for a while. I don't expect to continue work on it for at least another week. #1047 and #775 are higher-priority and will take me a while.

**Comment by kgodey** [comment]
## Backend priorities

The frontend is catching up fast to the backend, so we really need to get backend work for [Initial Data Types](https://github.com/centerofci/mathesar/milestone/11) completed and start on [Links between Tables](https://github.com/centerofci/mathesar/milestone/47) ASAP.

We divided up the issues as follows:

1. @dmos62:
    1. **Filtering**:
        1. https://github.com/centerofci/mathesar/issues/406
        2. https://github.com/centerofci/mathesar/issues/410
        3. https://github.com/centerofci/mathesar/issues/413
        4. https://github.com/centerofci/mathesar/issues/418
        5. https://github.com/centerofci/mathesar/issues/427
        6. https://github.com/centerofci/mathesar/issues/431
        7. Reopen https://github.com/centerofci/mathesar/issues/388 if needed to make sure it works with the new filtering setup.
    1. **Data Type API updates:**:
        1. https://github.com/centerofci/mathesar/issues/960
        2. https://github.com/centerofci/mathesar/issues/1036
        3. https://github.com/centerofci/mathesar/issues/1037
2. @mathemancer:
    1. https://github.com/centerofci/mathesar/issues/845 + associated work
    2. **Grouping**:
        1. https://github.com/centerofci/mathesar/issues/408 
        2. https://github.com/centerofci/mathesar/issues/411 
        3. https://github.com/centerofci/mathesar/issues/414
        4. https://github.com/centerofci/mathesar/issues/384
        5. https://github.com/centerofci/mathesar/issues/429
        6. https://github.com/centerofci/mathesar/issues/419
        7. https://github.com/centerofci/mathesar/issues/432 (currently blocked)
3. @silentninja:
    1. **Replace column name usage with ID:**
        1. https://github.com/centerofci/mathesar/issues/839
            1. https://github.com/centerofci/mathesar/issues/992
            2. https://github.com/centerofci/mathesar/issues/993
            3. https://github.com/centerofci/mathesar/issues/994
            4. https://github.com/centerofci/mathesar/issues/995
        2. #896    
    2. https://github.com/centerofci/mathesar/issues/1064
    3. https://github.com/centerofci/mathesar/issues/433
    4. Work coming out of https://github.com/centerofci/mathesar/issues/1063 (currently blocked)

I'll also work on #1008 if I have time this week.

@dmos62 @mathemancer @silentninja Please comment below with thoughts/concerns.

**Reply by dmos62** [reply]
No comment.

**Reply by kgodey** [reply]
@dmos62 Do we need to reopen https://github.com/centerofci/mathesar/issues/388 to make sure it works with the new filtering setup?

**Reply by dmos62** [reply]
@kgodey we can. I'm currently going through those filtering issues and creating test suites for filtering those types. I'll do the same for boolean when I get to it.

**Reply by mathemancer** [reply]
These priorities make sense to me.

**Reply by silentninja** [reply]
I will be working on 
- #433 
- #1064 
 
as there are a few things to discuss before starting work on #839.

**Comment by kgodey** [comment]
## Design priorities

1. #1063 is top priority since it's blocking backend work.
2. Continue exploratory work for Views in the design prototype.
3. Design specs for:
    1. #1065
    2. #978 

@ghislaineguerin @ppii775 Please comment below with thoughts/concerns

**Reply by ppii775** [reply]
@kgodey @ghislaineguerin apologies – i was under the weather for a few days. Hope to catch up in the coming days!

**Reply by ghislaineguerin** [reply]
@kgodey looks good

**Comment by kgodey** [comment]
## Product priorities

1. Finalize definition of Views, which will result in updates to the GitHub issues for Views design and the [product spec for Views](https://wiki.mathesar.org/en/product/specs/2022-01-views).

**Comment by kgodey** [comment]
## Comms Assignee schedule

| Week | Person |
|-|-|
| 2022-02-21 to 2022-02-25 | Ghislaine |
| 2022-02-28 to 2022-03-04 | Brent |
| 2022-03-07 to 2022-03-11 | Kriti |
| 2022-03-14 to 2022-03-18 | Pavish |
| 2022-03-21 to 2022-03-25 | Mukesh |
| 2022-03-28 to 2022-04-01 | Sean |
| 2022-04-04 to 2022-04-08 | Dominykas |

### Instructions
You can also find the schedule and instructions for the Comms Assignee on the [wiki page](https://wiki.mathesar.org/team/guide/comms-assignee). Please feel free to update the wiki page if you think of anything useful during your shift.

**Comment by kgodey** [comment]
## Upcoming Core Team Events

We're going to be playing https://skribbl.io/ during our next team event on March 11. It would be great to get some ideas for team events for the following dates:
- March 25
- April 8
- April 22

If your idea involves prep ahead of time for a host, also mention whether you want to host the event.

**Comment by dmos62** [comment]
>    How did your most recent work go?

Well. Steady progress.

>    What are you planning to work on soon?

Continue submiting PRs that add new filters. Then move on to the data type API issues.

>   Do you anticipate any dependencies or blockers that someone else on the team can help you with?

No.

**Comment by mathemancer** [comment]
## Brent's Check in

> How did your most recent work go?

Pretty well.  I'm really happy to be finishing up the last type implementation work (for now).  It was a bit more complicated than we originally thought (as usual), but the new implementation will be more robust and less reliant on python's parsing of both strings and postgres types into the same python type values.

> What are you planning to work on soon?

Once my current Date/Time type implementation is merged, I'm into grouping.  I'm also updating the date / time spec as I refine the implementation, and hope to get that merged into the wiki (though that's less urgent).

> Do you anticipate any dependencies or blockers that someone else on the team can help you with?

It seems like consensus is developing around the "human-readable dates and times" spec, but it still requires some coordination and participation to get that merged.  I also suspect we're going to need to coordinate on the column ID discussion for @silentninja 's work.

**Comment by pavish** [comment]
## Pavish's check in

> How did your most recent work go?

It went well, progress is steady. I'm working on text and boolean mathesar type implementation which involves some groundwork.

> What are you planning to work on soon?

I under-estimated the time when setting last week's goals, so the goals for this week remains mostly the same.

* Get PR https://github.com/centerofci/mathesar/pull/1080 merged.
* https://github.com/centerofci/mathesar/issues/257
* https://github.com/centerofci/mathesar/issues/970

> Do you anticipate any dependencies or blockers that someone else on the team can help you with?

None

**Comment by ghislaineguerin** [comment]
> How did your most recent work go?

Went well. Proposed change to design process (lower fidelity wireframes) by Kriti is speeding up design exploration.

> What are you planning to work on soon?

#1065 

> Do you anticipate any dependencies or blockers that someone else on the team can help you with?

None

**Comment by silentninja** [comment]
## Mukesh's Check in

> How did your most recent work go?

Made good progress on issues I am working on. Synchronous weekly check meetings were quite helpful in resolving discussions.

> What are you planning to work on soon?

I am planning to work on #1064 followed by #839

> Do you anticipate any dependencies or blockers that someone else on the team can help you with?

[Discussion on replacing column index](https://github.com/centerofci/mathesar/discussions/1081) is blocking work on #839.

**Comment by seancolsen** [comment]
## Sean's check in

> How did your most recent work go?

- Going pretty well. I feel good about where we're at with the E2E testing now.

> What are you planning to work on soon?

- The refactoring I'm doing to support column ids instead of names for #1047 has become a bit more intensive than I originally thought. I'm hoping to finish it tomorrow.
- After that, #775 is my next priority, and I expect that work to carry into next week.
- #1076, though in draft state, is on hold for a while while I focus on higher-priority work.

> Do you anticipate any dependencies or blockers that someone else on the team can help you with?

- Nope

**Comment by kgodey** [comment]
## Kriti's check in

> How did your most recent work go?

Pretty well, got a lot of work done on the [product spec for Views](https://wiki.mathesar.org/en/product/specs/2022-01-views). I think it is close to another round of review, I just need to work on specifying formulas more.

> What are you planning to work on soon?

- Complete work on formulas for the Views product spec
- Open a new discussion for Views product spec review
- Update design tickets for Views based on product spec
- #1008 if time permits

> Do you anticipate any dependencies or blockers that someone else on the team can help you with?

I'll need the product team to look at the Views discussion once I have it up in the next couple of days. @mathemancer @pavish @dmos62 @silentninja.
